### ISSUE TITLE
`segfault::segfault()` should be marked as `-> !` since they do not return

### ISSUE THREAD

**Issue by ExoSkye** [Vendicated/segfault-rs#4]
I believe, this may make it easier for the compiler to optimise the segfault, otherwise, it is too slow for my purposes.

**Comment by Oderjunkie** [comment]
+1. currently my function is failing to typecheck because of this

**Comment by ExoSkye** [comment]
Can confirm #5 fixes this, it allows the compiler to optimise more, if needed I can provide proof that this allows it to segfault faster

I would recommend that #5 is merged ASAP, otherwise it will become a fairly large issue for me

Kind Regards,
Skye

**Comment by ExoSkye** [comment]
I couldn't wait to provide this proof, since the implications of this are groundbreaking:

It reduced the size of my test app from 4248304 bytes to 4242464 bytes (a whopping 5.84KB difference!)
It also reduced the runtime of my test application from 49.21ms to 47.32ms (that's a decrease of 4%!!)

I hope you agree that these finding are conclusive and have wide reaching implications. Including applications such as the common `foo bar baz` app, alongside other high performance applications such as `Hello World` apps.

Thank you for your time,
Skye

**Comment by MeguminSama** [comment]
actually quite incredible

**Comment by oshaboy** [comment]
I put it into godbolt on no optimization 
Current:
```
  example::main:
          push    rax
          call    qword ptr [rip + example::segfault@GOTPCREL]
          pop     rax
          ret
```
With unreachable: 
```
  example::main:
          push    rax
          mov     rax, qword ptr [rip + example::segfault@GOTPCREL]
          call    rax
          ud2
```

So it should be around the same speed. The same code is generated for both the reachable and the unreachable. in any other optimization level. 

However do anything else after the main() function and the savings become apparent. Less code means better data locality and thus more speed.

**Comment by ExoSkye** [comment]
This was our point, in a large codebase there is likely to be decent amounts of code after the segfault call, so this doesn't change the severity of this issue (it is highly severe)

**Comment by Vendicated** [comment]
A fix for this was just released in v0.1.1 thanks to @Oderjunkie's inspiring contribution
